brainscore_vision/model_helpers/activations/core.py

```diff
@@ -231,6 +379,115 @@ def insert_attrs(self, wrapper):
         wrapper.register_batch_activations_hook = self.register_batch_activations_hook
         wrapper.register_stimulus_set_hook = self.register_stimulus_set_hook
 
+    def select_microsaccade(self, image_path: str, trial_number: int, image_shape: Tuple[int, int]
+                            ) -> Tuple[float, float]:
+        """
+        A function for generating microsaccade locations. The function returns a tuple of pixel shifts expanding from
+        the center of the image.
+
+        Microsaccade locations are placed within a circle, evenly distributed across the entire area in a spiral,
+        from the center to the circumference. We keep track of microsaccades both on a pixel and visual angle basis,
+        but only pixel values are returned. This is because shifting the image using cv2 requires pixel representation.
+        """
+        # avoid duplicating this computation repeatedly
+        if image_path in self.microsaccades.keys():
+            return self.microsaccades['pixels'][image_path][trial_number]
+
+        if image_shape[0] != image_shape[1]:
+            self._logger.debug('Warning: input image is not a square. Image dimension 0 is used to calculate the '
```

Comment:
is this a warning or debug information? :) there is `self._logger.warning` for more serious warnings

Reply:
debug information. fixed